### PR TITLE
Final touches

### PR DESCRIPTION
**Pull request by lukechampine**
I'm not too happy with the functions in `revise.go`. They could definitely use some refactoring...later.

### CODE REVIEW

File: modules/renter/contractor/editor.go [hunk 1]
```diff
@@ -104,75 +152,49 @@ func (he *hostEditor) Upload(data []byte) (crypto.Hash, error) {
...
-	var newRoots []crypto.Hash
+	// calculate price
+	sectorPrice, sectorCollateral := types.ZeroCurrency, types.ZeroCurrency
+
+	// calculate the new Merkle root
+	newRoots := make([]crypto.Hash, 0, len(he.contract.MerkleRoots))
 	index := -1
 	for i, h := range he.contract.MerkleRoots {
-		if h != root {
+		if h == root {
```
Comment (lukechampine) [inline modules/renter/contractor/editor.go]: derp

File: modules/renter/contractor/editor.go [hunk 2]
```diff
@@ -181,42 +203,68 @@ func (he *hostEditor) Delete(root crypto.Hash) error {
...
 		SectorIndex: uint64(index),
-	}})
-	if err != nil {
-		return err
+	}}
+	rev := newRevision(he.contract.LastRevision, merkleRoot, uint64(len(newRoots)), sectorPrice, sectorCollateral)
+
+	// run the revision iteration
+	return he.runRevisionIteration(actions, rev, newRoots, height)
+}
```
Comment (DavidVorick) [inline modules/renter/contractor/editor.go]: interesting that you would have each action be a separate function, that's going to make batching more difficult later

Comment (DavidVorick) [inline modules/renter/contractor/editor.go]: oh nevermind I see how this is structured

File: modules/renter/contractor/editor.go (hunk 1 above)
Comment (DavidVorick) [inline modules/renter/contractor/editor.go]: lol. unit tests ftw

**Pull request merged**